### PR TITLE
[Bug] Fix bug push value predicate of unique table when have sequence column

### LINKED ISSUE #10077: [Bug] Push down value predicate in unique table base tablet with sequence column may cause error result
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

unique table have version [0-1] [2-Y]
if rowset version [2-Y] is not compactioned means different segment may have same key like:
segment 1                    segment 2 
k1, v1, seq1                 k1, v2, seq2

seq2 > seq1. so we need output the `k1,v2`

but if the value predicate is `v = v1`, we get wrong result of `k1, v1`. because the `k1,v2,seq2` is filter by the `value predicate`

### What You Expected?

right result

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by HappenLee**
# Proposed changes

Issue Number: close #10077 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (Yes)
4. Does it need to update dependencies: (Yes)
5. Are there any changes that cannot be rolled back: (Yes)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/beta_rowset_reader.h [hunk 1]
```diff
@@ -53,6 +53,8 @@ class BetaRowsetReader : public RowsetReader {
     RowsetTypePB type() const override { return RowsetTypePB::BETA_ROWSET; }
 
 private:
+    bool _should_push_down_value_predicates() const;
```
Comment (yangzhg) [inline be/src/olap/rowset/beta_rowset_reader.h:56]: private function should not  with prefix  `_`

**Review by yangzhg — DISMISSED** [review]
LGTM

File: be/src/olap/rowset/beta_rowset_reader.cpp [hunk 2]
```diff
@@ -237,4 +235,12 @@ Status BetaRowsetReader::next_block(vectorized::Block* block) {
     return Status::OK();
 }
 
+bool BetaRowsetReader::_should_push_down_value_predicates() const {
```
Comment (Gabriel39) [inline be/src/olap/rowset/beta_rowset_reader.cpp:238]: should we check if there is a sequence column in tablet schema?

**Comment by morningman** [comment]
please create an issue to describe the problem or error msg

File: be/src/olap/rowset/beta_rowset_reader.cpp (hunk 2 above)
Comment (HappenLee) [reply be/src/olap/rowset/beta_rowset_reader.cpp:238]: not need. `not_overlapping()` means the rowset is compactioned, after compactioned the sequence column have same key only keep one row

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**